File: src/main/scala/com/wikibooks/spark/ch5/scala/HiveSample.scala
package com.wikibooks.spark.ch5.scala

import org.apache.spark.sql.{SaveMode, SparkSession}

// 5.6절 하이브 연동
object HiveSample {

  def main(args: Array[String]) = {

    val spark = SparkSession.builder()
      .appName("Spark Hive Example")
      .master("local[*]")
      .config("spark.sql.warehouse.dir", "/Users/beginspark/hive/warehouse")
      .enableHiveSupport()
      .getOrCreate()

    import spark.implicits._
    import spark.sql

    val row1 = Person("hayoon", 7, "student")
    val row2 = Person("sunwoo", 13, "student")
    val row3 = Person("hajoo", 5, "kindergartener")
    val row4 = Person("jinwoo", 13, "student")
    val data = List(row1, row2, row3, row4)
    val ds = spark.createDataset(data)

    sql("CREATE TABLE IF NOT EXISTS Persons (name STRING, age INT, job STRING)")
    spark.sql("show tables").show

    ds.toDF().write.mode(SaveMode.Overwrite).saveAsTable("Users")
    spark.sql("show tables").show

    spark.stop
  }
}

File: src/main/java/com/wikibooks/spark/ch5/HiveSample.java
package com.wikibooks.spark.ch5;

import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Encoders;
import org.apache.spark.sql.SaveMode;
import org.apache.spark.sql.SparkSession;

import java.util.Arrays;
import java.util.List;

// 5.6절 하이브 연동
public class HiveSample {

  public static void main(String[] args) {

    SparkSession spark = SparkSession
            .builder()
            .appName("Spark Hive Example")
            .master("local[*]")
            .config("spark.sql.warehouse.dir", "file:///Users/beginspark/hive/warehouse")
            .enableHiveSupport()
            .getOrCreate();

    JavaSparkContext sc = new JavaSparkContext(spark.sparkContext());

    Person row1 = new Person("hayoon", 7, "student");
    Person row2 = new Person("sunwoo", 13, "student");
    Person row3 = new Person("hajoo", 5, "kindergartener");
    Person row4 = new Person("jinwoo", 13, "student");

    List<Person> data = Arrays.asList(row1, row2, row3, row4);
    Dataset<Person> ds = spark.createDataset(data, Encoders.bean(Person.class));

    spark.sql("CREATE TABLE IF NOT EXISTS Persons (name STRING, age INT, job STRING)");
    spark.sql("show tables").show();

    ds.toDF().write().mode(SaveMode.Overwrite).saveAsTable("Users");
    spark.sql("show tables").show();

    spark.stop();
  }
}
